### ISSUE TITLE
Support PATCH as a valid HttpOption method

### ISSUE THREAD

**Issue by mbark** [isaacplmann/ngrx-query#4]
Currently the supported methods for HttpOptions are: `'DELETE' | 'GET' | 'HEAD' | 'POST' | 'PUT'`, however `PATCH` and `OPTIONS` are also a [valid methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) and it would be nice if the the method field in `HttpOptions` supported them. Right now I'm getting around the problem by simply bypassing the typing with `method: <any>'PATCH'`.

**Comment by isaacplmann** [comment]
You're right those should be added.  I'd accept a PR if you want to add that to the types.
